cohort-engine/src/main/java/com/ibm/cohort/engine/cos/CosDaoFactory.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * (C) Copyright IBM Corp. 2021, 2021
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.cohort.engine.cos;
+
+import com.ibm.cloud.objectstorage.ClientConfiguration;
+import com.ibm.cloud.objectstorage.auth.AWSCredentials;
+import com.ibm.cloud.objectstorage.auth.AWSStaticCredentialsProvider;
+import com.ibm.cloud.objectstorage.client.builder.AwsClientBuilder;
+import com.ibm.cloud.objectstorage.oauth.BasicIBMOAuthCredentials;
+import com.ibm.cloud.objectstorage.services.s3.AmazonS3;
+import com.ibm.cloud.objectstorage.services.s3.AmazonS3ClientBuilder;
+
+public class CosDaoFactory {
+
+	private CosDaoFactory() {
+	}
+
+	public static CosDao from(CosConfiguration configuration) {
+		AmazonS3 cosClient = createClient(
+				configuration.getApiKey(),
+				configuration.getServiceInstanceId(),
+				configuration.getEndpointUrl(),
+				configuration.getLocation()
+		);
+
+		return new CosDao(cosClient);
+	}
+
+	private static AmazonS3 createClient(String apiKey, String serviceInstanceId, String endpointUrl, String location) {
+		AWSCredentials credentials = new BasicIBMOAuthCredentials(apiKey, serviceInstanceId);
+		ClientConfiguration clientConfig = new ClientConfiguration()
+				.withRequestTimeout(5_000)
+				.withTcpKeepAlive(true);
```

Comment:
We should be able to move these to a config file or something.
Nothing immediately needed, but just want to log my thoughts.

Reply:
@CoreySandersOH @dkwasny-ibm 
_re:_
> Where the code lives

After thinking about this a bit more, 
I do agree that it should be it's in a separate module for the same reasons you two are talking about.

I guess my question would be should I also move parts of the caching interface code as well at that point?